ceph/ceph_admin/bootstrap.py

```diff
@@ -52,22 +55,123 @@ def construct_registry(cls, registry: str, json_file: bool = False):
     return config_dict_to_string(reg_args)
 
 
-class BootstrapMixin:
-    """Add bootstrap support to the child class."""
+def validate_fsid(cls, fsid: str, out: str) -> bool:
+    """
+    Method is used to validate fsid
+    Args:
+        cls: class object
+        fsid: <ID>
+        out: bootstrap response logs
 
-    def get_method(self, key):
-        """
-        Fetch method using the Key
-        Args:
-            key: fsid
+    Returns: Boolean value whether fsid is valid
+
+    """
+    out, err = cls.shell(args=["ceph", "fsid"])
+    return out == fsid
 
-        Returns: Returns the method based on the Key
 
-        """
-        verify_map = {
-            "fsid": self.validate_fsid,
-        }
-        return verify_map.get(key)
+def validate_dashboard(cls, out, user=None, password=None):
+    """
+    Method to validate dashboard login using provided user/password
+    and Also validates user/passwd in bootstrap console output
+
+    Args:
+        cls: class object
+        out: bootstrap response log
+        user: dashboard user
+        password: dashbard password
+
+    Returns:
+        boolean
+    """
+    _, _, db_log_part = out.partition("Ceph Dashboard is now available at:")
+
+    if not db_log_part:
+        logger.error("Dashboard log part not found")
+        return False
+
+    def parse(string):
+        str_ = r"%s:\s(.+)" % string
+        return re.search(str_, db_log_part).group(1)
+
+    host = parse("URL")
+    user_ = parse("User")
+    passwd_ = parse("Password")
+
+    if user:
+        assert user == user_, f"user {user} did not match"
+    if password:
+        assert password == passwd_, f"password {password} did not match"
+
+    host_ = re.search(r"https?:[/]{2}(.*):8443", host).group(1)
+    for node in cls.cluster.get_nodes():
+        if host_ in node.shortname:
```

Comment:
What is the probability of this check failing when we use `--allow-fqdn-hostname`?

Reply:
with current CephVMNode, there is no issue. 
However It will be problem if,
- hostname and FQDN are different.
- hostname is not substring of FQDN.